lib/urlwatch/reporters.py

```diff
@@ -935,44 +938,129 @@ def _format_output(self, job_state):
+                device_name = input(f"Choose a name for this device: [{self.DEFAULT_DEVICE_NAME}] ")
+                if not device_name:
+                    device_name = self.DEFAULT_DEVICE_NAME
+                pw = getpass.getpass(f"Matrix password for {self.user} (it will not be stored): ")
+                resp = await super().login(pw, device_name=device_name)
+                if isinstance(resp, nio.LoginResponse):
+                    logger.debug(f'Saving credentials to "{credentials_file}"')
+                    with credentials_file.open('w') as f:
+                        credentials = {
+                            'user_id': self.user_id,
+                            'device_id': self.device_id,
+                            'access_token': self.access_token
+                        }
+                        json.dump(credentials, f)
+                else:
+                    raise Exception('Matrix login failed, wrong user/password?')
+
+
 class MatrixReporter(MarkdownReporter):
     """Send a message to a room using the Matrix protocol"""
     MAX_LENGTH = 16384
 
     __kind__ = 'matrix'
 
     def submit(self):
-        if matrix_client is None:
-            raise ImportError('Python module "matrix_client" not installed')
-
-        homeserver_url = self.config['homeserver']
-        access_token = self.config['access_token']
-        room_id = self.config['room_id']
+        if nio is None:
+            raise ImportError('Python module "matrix-nio" not installed')
 
         body_markdown = '\n'.join(super().submit(MatrixReporter.MAX_LENGTH))
 
         if not body_markdown:
             logger.debug('Not calling Matrix API (no changes)')
             return
 
-        client_api = matrix_client.api.MatrixHttpApi(homeserver_url, access_token)
-
+        content = {
+            "msgtype": "m.text",
+            "body": body_markdown,
+        }
         if Markdown is not None:
-            body_html = Markdown(extras=["fenced-code-blocks", "highlightjs-lang"]).convert(body_markdown)
-
-            client_api.send_message_event(
-                room_id,
-                "m.room.message",
-                content={
-                    "msgtype": "m.text",
-                    "format": "org.matrix.custom.html",
-                    "body": body_markdown,
-                    "formatted_body": body_html
-                }
-            )
+            body_html = Markdown(
+                extras=["fenced-code-blocks", "highlightjs-lang"]).convert(
+                    body_markdown)
+
+            content["format"] = "org.matrix.custom.html"
+            content["formatted_body"] = body_html
         else:
-            logger.debug('Not formatting as Markdown; dependency on markdown2 not met?')
-            client_api.send_message(room_id, body_markdown)
+            logger.debug(
+                'Not formatting as Markdown; dependency on markdown2 not met?')
```

Comment:
I know this log message has been like this before, but since we touch it, maybe we can just make it something like:

    markdown2 could not be imported, skipping Markdown formatting